Shorten and smooth the transfer/multi-task/few-shot section (#117)
jaybee84/ml-in-rd: content/05.prior-knowledge.md

Transfer learning is an approach where a model trained for one task or domain (source domain) is applied to another, typically related task or domain (target domain).
Transfer learning can be supervised (one or both of the source and target domains have labels), or unsupervised (both domains are unlabeled).

> Though there are multiple types of transfer learning, in a later section we will focus in-depth on feature-representation-transfer [@doi:10.1109/TKDE.2009.191].

Reviewer: ```suggestion
Though there are multiple types of transfer learning, in a later section we will focus in-depth on feature-representation-transfer.
```
Not sure if necessary to cite twice here

Feature-representation-transfer approaches learn representations from the source domain and apply them to a target domain [@doi:10.1109/TKDE.2009.191].

> For example, low-dimensional representations can be learned from tumor transcriptomic data and transferred to describe patterns associated with genetic alterations in cell line data [@doi:10.1186/s13059-020-02021-3].  <!--TODO: anything need to be added/revised here once the dimensionality reduction section is rewritten to be more general? -->

Reviewer: ```suggestion
For example, low-dimensional representations can be learned from tumor transcriptomic data and transferred to describe patterns associated with genetic alterations in cell line data [@doi:10.1186/s13059-020-02021-3].  
<!--TODO: anything need to be added/revised here once the dimensionality reduction section is rewritten to be more general? -->
```
Minor suggestion but splitting this out to a new line so it's less likely to be missed :)

> Where transfer learning can be supervised or unsupervised, the related approaches multitask and few-shot learning are forms of supervised learning that generally rely on deep neural networks.

Reviewer: ```suggestion
Whereas transfer learning can be supervised or unsupervised, the related approaches multitask and few-shot learning are forms of supervised learning that generally rely on deep neural networks.
```

> Multitask learning is an approach where classifiers are learned for _related tasks_ at the same time using a shared representation [@doi:10.1023/A:1007379606734], where task refers to an individual prediction being made.

Reviewer: ```suggestion
Multitask learning is an approach where classifiers are learned for _related individual predictions_ (tasks) at the same time using a shared representation [@doi:10.1023/A:1007379606734].
```

Few-shot learning is the generalization of a model trained on related tasks to a new task with limited labeled data (e.g., the detection of a patient with a rare disease from a low number of examples of that rare disease).
Multitask and few-shot learning are comprised of a variety of approaches and architectures that are beyond this scope of this work (see [@arxiv:1706.05098; @arxiv:1707.08114v2] and [@arxiv:1904.05046v3] for an overview).
We include brief descriptions of selected studies to illustrate potential uses and limitations of these approaches below.

> Multitask neural networks that predict multiple tasks simultaneously are generally thought to improve performance over models that make predictions for a single task by learning a shared representation and effectively being exposed to more training data than the single task case [@doi:10.1023/A:1007379606734; @arxiv:1606.08793].

Reviewer: ```suggestion
Multitask neural networks (which predict multiple tasks simultaneously) are thought to improve performance over single task models by learning a shared representation, effectively being exposed to more training data than single task models [@doi:10.1023/A:1007379606734; @arxiv:1606.08793].
```

Kearnes, Goldman, and Pande set out to examine the effects of dataset size and task relatedness on multitask learning performance improvements ("multitask effect") in drug discovery–an area that also suffers from insufficient data [@arxiv:1606.08793].
The authors found that the multitask performance gains were highly dataset-specific: smaller datasets tended to benefit most from multitask learning and the addition of more training data did not guarantee improved performance for multitask models.
Ding et al. also demonstrated that performance gains were context-dependent; multitask neural networks outperformed single-task networks for predicting complex rare phenotypes from EHR data, but not common phenotypes [@arxiv:1808.03331].

> In contrast, one-shot or few-shot learning relies on using prior knowledge to generalize to new prediction tasks where there are a low number of examples [@arxiv:1904.05046v3], where a distance metric is learned from input data and used to compare new examples for prediction [@doi:10.1021/acscentsci.6b00367].

Reviewer: ```suggestion
In contrast, one-shot or few-shot learning relies on using prior knowledge to generalize to new prediction tasks [@arxiv:1904.05046v3], where a distance metric is learned from input data and used to compare new examples for prediction [@doi:10.1021/acscentsci.6b00367].
```

Altae-Tran et al. developed a method for predicting small molecule activity that learned a meaningful distance metric over the properties of various compounds [@doi:10.1021/acscentsci.6b00367].

> The authors' results suggested underperformance of one-shot learning methods relative to baseline random forest models when structural similarity could not be exploited and did not show support for generalization of models trained on very different contexts from the target task.

Reviewer: ```suggestion
The authors' results suggested that when structural similarity could not be exploited, one-shot learning methods underperform relative to baseline random forest models, and that models trained on very different contexts from the target task did not generalize well.
```

Reviewer: I thought my suggestion would be shorter, but doesn't look like it

> Quellec et al. presented a few-shot learning approach for detecting rare pathologies in fundus photographs [@arxiv:1907.09449v3] that outperformed multitask learning, which suggested few-shot learning provides an advantage in contexts where predicting common conditions simultaneously results in a loss of performance [@arxiv:1907.09449v3].  

Reviewer: ```suggestion
Quellec et al. presented a few-shot learning approach for detecting rare pathologies in fundus photographs that outperformed multitask learning, suggesting that few-shot learning provides an advantage in contexts where predicting common conditions simultaneously results in a loss of performance [@arxiv:1907.09449v3].  
```
